inference/usage_tracking/persistent_queue.py

```diff
@@ -0,0 +1,103 @@
+import json
```

Comment:
general question: could we estimate performance implications - from what I remember we will be using that on bg thread only once for a while, compressing messages avoiding excessive growth, is that right?

Reply:
yes, we are dumping usage payload to the queue every configurable timeout (currently 10 seconds), and it's done outside of the main thread